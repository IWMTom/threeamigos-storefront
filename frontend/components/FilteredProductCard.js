import gql from "graphql-tag";
import { Query } from "react-apollo";
import { Link } from "../routes";
import formatMoney from "../lib/formatMoney";

const FilteredProductCard = ({ product }) => (
	<Link route="product" params={{ slug: product.slug }} key={product.id}>
		<a className={product.sale ? "product card sale" : "product card"}>
			{product.sale && <span className="product-sale">Sale 50%</span>}
			<div className="pure-g">
				<div className="pure-u-1-4">
					<img src={product.image_url} />
				</div>
				<div className="product-content pure-u-3-4">
					<h2>{product.name}</h2>
					<span>
						<i className={product.category.icon} />
						{product.category.name}
					</span>
					<span>
						<i className="ic_bookmark" />
						{product.brand.name}
					</span>
					<p>{product.description}</p>
					<aside>
						<i
							className={
								product.stock === 0
									? "badge badge-grey"
									: product.sale
									? "badge badge-pink"
									: "badge badge-green"
							}
						>
							{product.stock === 0 ? "Out of Stock" : "In Stock"}
						</i>

						<span>
							{product.sale && (
								<del>{formatMoney(product.price * 2)}</del>
							)}
							{formatMoney(product.price)}
						</span>
					</aside>
				</div>
			</div>
		</a>
	</Link>
);

export default FilteredProductCard;
