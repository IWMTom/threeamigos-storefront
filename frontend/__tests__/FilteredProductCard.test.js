import FilteredProductCard from "../components/FilteredProductCard";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";

const mockedProduct = {
	id: "abcdefghijklmnopqrst",
	slug: "abc-def",
	sale: true,
	name: "Abc Def",
	image_url: "http://placehold.it/100x100",
	price: "199",
	category: {
		icon: "ic_bookmark",
		name: "Screen Protectors"
	},
	brand: {
		name: "iStuff-R-Us"
	}
};

describe("<FilteredProductCard />", () => {
	it("renders and matches the snapshot", () => {
		const wrapper = shallow(
			<FilteredProductCard product={mockedProduct} />
		);

		expect(toJSON(wrapper)).toMatchSnapshot();
	});
});
