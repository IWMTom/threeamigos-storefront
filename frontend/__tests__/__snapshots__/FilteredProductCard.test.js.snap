// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilteredProductCard /> renders and matches the snapshot 1`] = `
<LinkRoutes
  key="abcdefghijklmnopqrst"
  params={
    Object {
      "slug": "abc-def",
    }
  }
  route="product"
>
  <a
    className="product card sale"
  >
    <span
      className="product-sale"
    >
      Sale 50%
    </span>
    <div
      className="pure-g"
    >
      <div
        className="pure-u-1-4"
      >
        <img
          src="http://placehold.it/100x100"
        />
      </div>
      <div
        className="product-content pure-u-3-4"
      >
        <h2>
          Abc Def
        </h2>
        <span>
          <i
            className="ic_bookmark"
          />
          Screen Protectors
        </span>
        <span>
          <i
            className="ic_bookmark"
          />
          iStuff-R-Us
        </span>
        <p />
        <aside>
          <i
            className="badge badge-pink"
          >
            In Stock
          </i>
          <span>
            <del>
              £3.98
            </del>
            £1.99
          </span>
        </aside>
      </div>
    </div>
  </a>
</LinkRoutes>
`;
